### ISSUE TITLE
lora微调训练loss震荡

### ISSUE THREAD

**Issue by GSK666** [aigc-apps/CogVideoX-Fun#47]
请问一下，我用自己的视频数据在8卡A800机器上进行5B模型的lora微调，我调小了学习率但是发现训练loss一直在震荡，以下是我的微调参数和tensorboard记录，想请教一下这种情况是什么原因导致，后续应该如何优化：
accelerate launch --use_deepspeed --deepspeed_config_file config/zero_stage2_config.json --deepspeed_multinode_launcher standard scripts/train_lora.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --train_data_meta=$DATASET_META_NAME \
  --image_sample_size=1024 \
  --video_sample_size=1024 \
  --token_sample_size=512 \
  --video_sample_stride=3 \
  --video_sample_n_frames=49 \
  --train_batch_size=16 \
  --video_repeat=1 \
  --gradient_accumulation_steps=1 \
  --dataloader_num_workers=8 \
  --num_train_epochs=50 \
  --checkpointing_steps=500 \
  --learning_rate=1e-06 \
  --lr_scheduler="cosine" \
  --seed=42 \
  --output_dir="output_dir" \
  --gradient_checkpointing \
  --mixed_precision="bf16" \
  --adam_weight_decay=3e-2 \
  --adam_epsilon=1e-10 \
  --vae_mini_batch=1 \
  --max_grad_norm=0.05 \
  --random_hw_adapt \
  --training_with_video_token_length \
  --random_frame_crop \
  --enable_bucket \
  --use_deepspeed \
  --train_mode="inpaint" 
  
![0zm0y12000gayzzavF2A9](https://github.com/user-attachments/assets/5fc89da1-10a6-4fcf-8367-ac68ba2d9280)

**Comment by yunkchen** [comment]
这个100步的lora跑一下生成看看效果是否符合预期

**Comment by GSK666** [comment]
目前我用1000步的lora测试I2V生成4s的视频，发现有一些视频前1-2s稳定但后2s就会出现混乱的片段

**Comment by yunkchen** [comment]
> > 这个100步的lora跑一下生成看看效果是否符合预期
> 
> 目前我用1000步的lora测试I2V生成4s的视频，发现有一些视频前1-2s稳定但后2s就会出现混乱的片段

check下训练数据，调大一点lr试试； 
这种得看具体训练任务来分析

**Comment by GSK666** [comment]
> > > 这个100步的lora跑一下生成看看效果是否符合预期
> > 
> > 
> > 目前我用1000步的lora测试I2V生成4s的视频，发现有一些视频前1-2s稳定但后2s就会出现混乱的片段
> 
> check下训练数据，调大一点lr试试； 这种得看具体训练任务来分析

好的，非常感谢，麻烦再请教一下lora微调的数据量大概需要什么量级？

**Comment by yunkchen** [comment]
> > > > 这个100步的lora跑一下生成看看效果是否符合预期
> > > 
> > > 
> > > 目前我用1000步的lora测试I2V生成4s的视频，发现有一些视频前1-2s稳定但后2s就会出现混乱的片段
> > 
> > 
> > check下训练数据，调大一点lr试试； 这种得看具体训练任务来分析
> 
> 好的，非常感谢，麻烦再请教一下lora微调的数据量大概需要什么量级？

如果是某一种绘画风格，可能100～200张图片就非常足够了

**Comment by llpspark** [comment]
@GSK666 你好，这个问题是否解决？我遇到了相同的问题，调学习率未果。

**Comment by GSK666** [comment]
> @GSK666 你好，这个问题是否解决？我遇到了相同的问题，调学习率未果。

调整学习率和其他参数暂时没有解决，目前正在重新清洗训练数据，后续用质量更高的数据再进行测试

**Comment by bubbliiiing** [comment]
I just tried it and it can fit normally using this demo.
https://pai-aigc-photog.oss-cn-hangzhou.aliyuncs.com/easyanimate/asset/v2/Minimalism.zip

**Comment by GSK666** [comment]
> I just tried it and it can fit normally using this demo. https://pai-aigc-photog.oss-cn-hangzhou.aliyuncs.com/easyanimate/asset/v2/Minimalism.zip

Thanks. Maybe this problem is due to the fact that my training data is videos and the styles of the videos may not be uniform. I will try again with a higher quality video dataset later.

**Comment by ybiu** [comment]
> I just tried it and it can fit normally using this demo. https://pai-aigc-photog.oss-cn-hangzhou.aliyuncs.com/easyanimate/asset/v2/Minimalism.zip

I tried to use this data set to reproduce the experiment with the following settings, but the loss was also oscillating, and the generated video was almost the same as using lora. What is the problem?Please advise and thank you for your reply.
![2ce33c95611804970d59ba19288712a](https://github.com/user-attachments/assets/c19e1177-ead3-485d-9de3-1c469c3bcee5)

**Comment by ybiu** [comment]
> > I just tried it and it can fit normally using this demo. https://pai-aigc-photog.oss-cn-hangzhou.aliyuncs.com/easyanimate/asset/v2/Minimalism.zip
>
> I tried to use this data set to reproduce the experiment with the following settings, but the loss was also oscillating, and the generated video was almost the same as using lora. What is the problem?Please advise and thank you for your reply. ![2ce33c95611804970d59ba19288712a](https://private-user-images.githubusercontent.com/45233816/3927692 30-c19e1177-ead3-485d-9de3-1c469c3bcee5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRod WIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMzOTQ1NzcsIm5iZiI 6MTczMzM5NDI3NywicGF0aCI6Ii80NTIzMzgxNi8zOTI3NjkyMzAtYzE5ZTExNzctZWFkMy00ODVkLTlkZTMtMWM0NjljM2JjZWU1L nBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJT JGMjAyNDEyMDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA1VDEwMjQzN1omWC1BbXot RXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhNmQ5NzBiZjMxY2FkMWIyZDdkMTgyMGY5NTAyOTM1MThmYjI4YmFkZDMxZTQ1Mz VmMzE4OTk0NzJkYjdiODYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.b-MNXUCMzTf3uZz6y_O9XMtSe9d1XqbLa_uGbnFpMbw)

This is the setup for my experiment：
  export MODEL_NAME="models/Diffusion_Transformer/CogVideoX-Fun-V1.1-2b-InP"
export DATASET_NAME="datasets/Minimalism/"
export DATASET_META_NAME="datasets/Minimalism/metadata_add_width_height.json"
export NCCL_IB_DISABLE=1
export NCCL_P2P_DISABLE=1
NCCL_DEBUG=INFO

# When train model with multi machines, use "--config_file accelerate.yaml" instead of "--mixed_precision='bf16'".
accelerate launch --use_deepspeed --deepspeed_config_file config/zero_stage2_config.json --deepspeed_multinode_launcher standard scripts/train_lora.py \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --train_data_dir=$DATASET_NAME \
  --train_data_meta=$DATASET_META_NAME \
  --image_sample_size=1280 \
  --video_sample_size=256 \
  --token_sample_size=512 \
  --video_sample_stride=3 \
  --video_sample_n_frames=49 \
  --train_batch_size=8 \
  --video_repeat=1 \
  --gradient_accumulation_steps=1 \
  --dataloader_num_workers=8 \
  --num_train_epochs=100 \
  --checkpointing_steps=50 \
  --learning_rate=1e-03 \
  --seed=42 \
  --output_dir="output_dir_mini_1e03" \
  --gradient_checkpointing \
  --adam_weight_decay=3e-2 \
  --adam_epsilon=1e-10 \
  --vae_mini_batch=1 \
  --max_grad_norm=0.05 \
  --random_hw_adapt \
  --training_with_video_token_length \
  --random_frame_crop \
  --enable_bucket \
  --use_deepspeed \
  --low_vram \
  --train_mode="inpaint"
